File: 0769-max-chunks-to-make-sorted/0769-max-chunks-to-make-sorted.cpp
class Solution {
public:
    int maxChunksToSorted(vector<int>& arr) {
        int n = arr.size(), ans = 0;
        for(int i=0, sum = 0, index_sum; i<n; i++) {
            sum += arr[i];
            index_sum += i;
            if(index_sum == sum) ans++;
        }
        return ans;
    }
};

File: 0769-max-chunks-to-make-sorted/0769-max-chunks-to-make-sorted.java
class Solution {
    public int maxChunksToSorted(int[] arr) {
        int n = arr.length;
        int []left = new int[n]; 
        int []right = new int[n];
        left[0] = arr[0];
        for(int i=1; i<n; i++) {
            left[i] = Math.max(arr[i], left[i-1]);
        }
        right[n-1] = arr[n-1];
        for(int i=n-2; i>=0; i--) {
            right[i] = Math.min(arr[i], right[i+1]);
        }
        int ans = 1;
        for(int i=1; i<n; i++) {
            if(left[i-1] <= right[i]) ans++;
        }
        return ans;
    }
}